packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/listener/l7/create/controller.js

```diff
@@ -0,0 +1,73 @@
+export default class OctaviaLoadBalancerCreateL7PolicyCtrl {
+  /* @ngInject */
+  constructor(OctaviaLoadBalancerL7Service, Alerter, $translate) {
+    this.OctaviaLoadBalancerL7Service = OctaviaLoadBalancerL7Service;
+    this.Alerter = Alerter;
+    this.$translate = $translate;
+  }
+
+  $onInit() {
+    this.loading = false;
+    this.model = {
+      name: '',
+      position: 1,
+      action: undefined,
+      redirectHttpCode: undefined,
+      redirectPool: undefined,
+      redirectPrefix: undefined,
+      redirectUrl: undefined,
+    };
+  }
+
+  submit() {
+    this.loading = true;
+    this.trackL7CreatePolicyAction('confirm');
+    this.OctaviaLoadBalancerL7Service.createL7Policy(
+      this.projectId,
+      this.region,
+      this.listenerId,
+      this.model.name,
+      this.model.position,
+      this.model.action,
+      this.model.redirectHttpCode,
+      this.model.redirectPool?.id,
+      this.model.redirectPrefix,
+      this.model.redirectUrl,
+    )
+      .then(async (response) => {
+        this.trackL7CreatePolicyPage('success');
+        await this.goBackToL7PoliciesList(true).then(() =>
+          this.Alerter.set(
+            'alert-success',
+            this.$translate.instant(
+              'octavia_load_balancer_create_l7_policy_success',
+              {
+                policy: this.model.name,
+                link: this.getL7RuleCreationLink(response),
+              },
+            ),
+            null,
+            'octavia.alerts.l7Policies',
+          ),
+        );
+      })
+      .catch((error) => {
+        this.trackL7CreatePolicyPage('error');
+        this.Alerter.error(
+          this.$translate.instant('octavia_load_balancer_global_error', {
+            message: error.data?.message,
+            requestId: error.headers('X-Ovh-Queryid'),
+          }),
+          'octavia.alerts.l7Policy.create',
+        );
```

Comment:
Seems to be defined in a lot of controllers. Maybe you should put it in a global message service to avoid duplicating the code. What do you think ?